First commit that creates a base build docker image that includes

ChrisPRobinson opened the pull request:
the ruby gems for sentinel as well as a make file to build the docker
image when needed.

ChrisPRobinson committed:
First commit that creates a base build docker image that includes
the ruby gems for sentinel as well as a make file to build the docker
image when needed.

colstrom commented on sentinel/cli/Gemfile:
```diff
@@ -0,0 +1,18 @@
+source (ENV['RUBYGEMS_MIRROR'] or 'https://rubygems.org')
+
+ruby '2.1.7'
```
We should remove these from Gemfiles.

colstrom commented on sentinel/daemon/Gemfile:
```diff
@@ -0,0 +1,35 @@
+source (ENV['RUBYGEMS_MIRROR'] or 'https://rubygems.org')
+
+ruby '2.1.7' # prevents hard-to-diagnose errors with bundle install
```
We should remove these from Gemfiles. The comment here (from the Sentinel source) is false. These lines actually cause more issues than they solve.

colstrom commented on Dockerfile:
```diff
@@ -0,0 +1,24 @@
+FROM helioncf/ubuntu-core
+RUN apt-get update && apt-get install -y \
+  nodejs \
+  git-core curl zlib1g-dev build-essential libssl-dev libreadline-dev libyaml-dev libsqlite3-dev sqlite3 libxml2-dev libxslt1-dev libcurl4-openssl-dev python-software-properties libffi-dev \
```
Minor stylistic thing here... can we have these one package per line, in sorted order? Makes merges easier to read.

ChrisPRobinson replied:
done, I also added comments on which components and where they came from.

colstrom commented on Dockerfile:
```diff
@@ -0,0 +1,24 @@
+      ruby2.1-dev \
+      ruby2.1-doc
+
+RUN gem2.1 install bundler
```
Also don't need `gem2.1` here. `gem` will be the standard ruby version. Disregard @shsu with regards to `ruby-switch`, since this container has only one version of Ruby. It's only needed if you have more than one (the first one you install will be the default otherwise).

ChrisPRobinson replied:
Yes, understood, updated.

ChrisPRobinson committed:
Update based on comments and missing components
1) Added nodejs as this was missed
2) Added python
3) Fixed make clean to work better
4) Consolidated steps into 4 intermmediate layers

shsu commented on Dockerfile:
```diff
@@ -0,0 +1,56 @@
+  python \
+  python-dev \
+  python3.5 \
+  python3.5-dev \
```
Missing code to symlink `python3` to `python3.5`

ChrisPRobinson replied:
Updated below, but I don't think this change matters given our priority switch

ChrisPRobinson committed:
Symlink python 3 to 3.5

shsu commented:
LGTM, merging it.
